tesseract/tesseract_common/cmake/tesseract_macros.cmake

```diff
@@ -23,6 +23,10 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 macro(tesseract_variables)
```

Comment:
Is this a new macro? I don't remember it.

Reply:
> > I am not sure how PCL does it but it should be similar. There is a global way to expose all symbols which may be what they are doing, but this approach is the preferred way. This was copied form rclcpp, which i copied from [GCC website](https://gcc.gnu.org/wiki/Visibility).
> 
> Do IDEs get confused? This isn't going to make code development painful is it?
> 
> Edit: Still reviewing. I guess it took this comment as meaning I was done with the review.

I don't believe so. If you want a class or global function to visible you just need to add TESSERACT_PUBLIC and if it is not supposed to be visible you add TESSERACT_LOCAL and that is pretty much it.

Reply:
> Is this a new macro? I don't remember it.

No it was added as part of the switch to cmake_common_scripts.

Reply:
So if visibility is turned off, then a target that links it will not be able to see it? Could we just default them to visible and set the ones we want private (since there are only a few). I assume not since that is basically what we had before just with none set not visible